ADAL/src/cache/ADTokenCacheItem+Internal.m

```diff
@@ -88,9 +88,23 @@ - (ADAuthenticationResult *)processTokenResponse:(NSDictionary *)response
     
     [self checkCorrelationId:response requestCorrelationId:requestCorrelationId];
     
-    ADAuthenticationError* error = [ADAuthenticationContext errorFromDictionary:response errorCode:(fromRefreshTokenWorkflow) ? AD_ERROR_SERVER_REFRESH_TOKEN_REJECTED : AD_ERROR_SERVER_OAUTH];
+    ADAuthenticationError* error = [ADAuthenticationContext errorFromDictionary:response errorCode:(refreshToken) ? AD_ERROR_SERVER_REFRESH_TOKEN_REJECTED : AD_ERROR_SERVER_OAUTH];
     if (error)
     {
+        if (refreshToken)
+        {
+            NSMutableDictionary *userInfo = [error userInfo] ? [[error userInfo] mutableCopy] : [[NSMutableDictionary alloc] initWithCapacity:1];
+            if (refreshToken.userInformation.userId)
+            {
+                [userInfo setObject:refreshToken.userInformation.userId forKey:ADUserIdKey];
+            }
+            error = [ADAuthenticationError errorWithDomain:error.domain
+                                                      code:error.code
+                                         protocolErrorCode:error.protocolCode
+                                              errorDetails:error.errorDetails
+                                             correlationId:requestCorrelationId
+                                                  userInfo:userInfo];
```

Comment:
Shall we have a function to handle the logic of adding entries to error userInfo? say `[ADAuthenticationError errorFromExistingError:additionalUserInfo:]`? :)